src/app/src/main/kotlin/org/imperial/mrc/hint/Extensions.kt

```diff
@@ -63,10 +63,10 @@ fun Response.asResponseEntity(logger: Log = LogFactory.getLog(HintExceptionHandl
 
         if (!json.has("status") && !json.has("success"))
         {
-            logger.error(json)
-            throw IOException("Response does not have status or success properties")
+            if (httpStatus < HttpStatus.BAD_REQUEST ) SuccessResponse(json).asResponseEntity()
+            else ErrorDetail(httpStatus, json.asText()).toResponseEntity<String>()
```

Comment:
But you're right about the logging, I missed that. Could log directly here, or rather than returning `ErrorDetail` throw some generic exception with the json response as the message...

Reply:
Ok, cool, thanks. In which case suggest something like this:
```suggestion
            if (!httpStatus.isError)
            {
                SuccessResponse(json).asResponseEntity()
            }
            else
            {
                val message = json.asText()
                logger.error(message)
                ErrorDetail(httpStatus, message).toResponseEntity<String>()
            }
```